tests/virtual_machines_test.go

```diff
@@ -434,7 +435,7 @@ var _ = Describe("[rfe_id:3503][crit:medium][vendor:cnv-qe@redhat.com][level:com
 					Eventually(func() error {
 						err = testClient.VirtClient.Create(context.TODO(), newVM)
 						if err != nil {
-							Expect(strings.Contains(err.Error(), "Failed to create virtual machine allocation error: the range is full")).To(Equal(true), "Should only get a range full error until cache get updated")
+							Expect(err).Should(MatchError(errors.New("Failed to create virtual machine allocation error: the range is full")), fmt.Sprintf("Should only get a range full error until cache get updated. got instead: %v", err))
```

Comment:
No need for "got instead" gomega should show you "obtained" and "expected" error.

Reply:
I thought so as well, but it doesn't:
```
• Failure [26.909 seconds]
[rfe_id:3503][crit:medium][vendor:cnv-qe@redhat.com][level:component]Virtual Machines
/home/ralavi/go/src/github.com/k8snetworkplumbingwg/kubemacpool/tests/virtual_machines_test.go:30
  Check the client
  /home/ralavi/go/src/github.com/k8snetworkplumbingwg/kubemacpool/tests/virtual_machines_test.go:63
    When Running with default opt-mode configuration
    /home/ralavi/go/src/github.com/k8snetworkplumbingwg/kubemacpool/tests/virtual_machines_test.go:88
      and a VM's NIC is removed and a new VM is created with the same MAC
      /home/ralavi/go/src/github.com/k8snetworkplumbingwg/kubemacpool/tests/virtual_machines_test.go:401
        [test_id:2995]should successfully release the MAC and the new VM should be created with no errors [It]
        /home/ralavi/go/src/github.com/k8snetworkplumbingwg/kubemacpool/tests/virtual_machines_test.go:402

        Should only get a range full error until cache get updated. got instead: <nil>
        Expected an error, got nil

        /home/ralavi/go/src/github.com/k8snetworkplumbingwg/kubemacpool/vendor/github.com/onsi/gomega/internal/assertion/assertion.go:28

```

Reply:
It is, since it's nill is no tshowing you the obtained.
```
Expected an error, got nil
```

Reply:
[Done](https://github.com/k8snetworkplumbingwg/kubemacpool/compare/63f301ee629d975cc9cf8b7b8a04da7f78f5acc1..44c9607de282decb04fbd2910f30912bdf57f677)